sdk/keyvault/keyvault-keys/src/cryptography/aesCryptographyProvider.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationOptions } from "@azure/core-http";
+import * as crypto from "crypto";
+import {
+  EncryptOptions,
+  EncryptResult,
+  DecryptOptions,
+  DecryptResult,
+  KeyWrapAlgorithm,
+  WrapKeyOptions,
+  WrapResult,
+  UnwrapKeyOptions,
+  UnwrapResult,
+  SignOptions,
+  SignResult,
+  VerifyOptions,
+  VerifyResult,
+  AesCbcEncryptParameters,
+  JsonWebKey
+} from "..";
+import { AesCbcDecryptParameters } from "../cryptographyClientModels";
+import {
+  CryptographyProvider,
+  CryptographyProviderOperation,
+  LocalCryptographyUnsupportedError
+} from "./models";
+
+/**
+ * An AES cryptography provider supporting AES algorithms.
+ * @internal
+ */
+export class AesCryptographyProvider implements CryptographyProvider {
+  private key: JsonWebKey;
+  constructor(key: JsonWebKey) {
+    this.key = key;
+  }
+  encrypt(
+    encryptParameters: AesCbcEncryptParameters,
+    _options?: EncryptOptions
+  ): Promise<EncryptResult> {
+    const { algorithm, keySizeInBytes } = this.supportedAlgorithms[encryptParameters.algorithm];
+    const iv = encryptParameters.iv || crypto.randomBytes(16);
+
+    this.ensureValid(keySizeInBytes);
+
+    const cipher = crypto.createCipheriv(algorithm, this.key.k!.subarray(0, keySizeInBytes), iv);
+    let encrypted = cipher.update(Buffer.from(encryptParameters.plaintext));
+    encrypted = Buffer.concat([encrypted, cipher.final()]);
+
+    return Promise.resolve({
+      algorithm: encryptParameters.algorithm,
+      result: encrypted,
+      iv: iv
+    });
+  }
+
+  decrypt(
+    decryptParameters: AesCbcDecryptParameters,
+    _options?: DecryptOptions
+  ): Promise<DecryptResult> {
+    const { algorithm, keySizeInBytes } = this.supportedAlgorithms[decryptParameters.algorithm];
+
+    this.ensureValid(keySizeInBytes);
+
+    const decipher = crypto.createDecipheriv(
+      algorithm,
+      this.key.k!.subarray(0, keySizeInBytes),
+      decryptParameters.iv
+    );
+    let dec = decipher.update(Buffer.from(decryptParameters.ciphertext));
+    dec = Buffer.concat([dec, decipher.final()]);
+
+    return Promise.resolve({
+      algorithm: decryptParameters.algorithm,
+      result: dec
+    });
+  }
+
+  isSupported(algorithm: string, operation: CryptographyProviderOperation): boolean {
+    if (!this.key.k) {
+      return false;
+    }
+
+    if (!Object.keys(this.supportedAlgorithms).includes(algorithm)) {
+      return false;
+    }
+
+    if (!this.supportedOperations.includes(operation)) {
+      return false;
+    }
+
+    return true;
+  }
+
+  /**
+   * The set of algorithms this provider supports.
+   * For AES encryption, the values include the underlying algorithm used in crypto
+   * as well as the key size in bytes.
+   *
+   * We start with support for A[SIZE]CBCPAD which uses the PKCS padding (the default padding scheme in node crypto)
```

Comment:
As far as I can tell there's really no way to set the padding mode, only disable auto-padding entirely

https://nodejs.org/docs/latest-v8.x/api/crypto.html#crypto_decipher_setautopadding_autopadding

Reply:
Based on the docs this seems okay: either no padding, or PKCS padding - assuming that's PKCS7 padding.

Reply:
Resolving - no padding is the right thing for AES-CBC. We just need a service-side issue to be addressed first. Will add an issue for me to add AES-CBC support locally when that's resolved.